connectors/byoei.py

```diff
@@ -214,7 +238,10 @@ def __str__(self):
         )
 
     async def run(self, generator):
-        await self.get_docs(generator)
+        try:
+            await self.get_docs(generator)
+        except asyncio.CancelledError:
+            logger.info("Task is canceled, stop Fetcher...")
```

Comment:
Do we need to have a `finally` block here that disposes of used resources/tasks?